lineman/app/components/thread_page/comment_form/comment_form.coffee

```diff
@@ -3,15 +3,16 @@ angular.module('loomioApp').directive 'commentForm', ->
   restrict: 'E'
   templateUrl: 'generated/components/thread_page/comment_form/comment_form.html'
   replace: true
-  controller: ($scope, FlashService, Records, CurrentUser) ->
+  controller: ($scope, FormService, Records, CurrentUser) ->
     group = $scope.comment.discussion().group()
     discussion = $scope.comment.discussion()
 
-    $scope.submit = ->
-      $scope.comment.save().then ->
-        $scope.comment = Records.comments.build(discussion_id: discussion.id)
-        $scope.$emit('commentSaveSuccess')
-        FlashService.success('comment_form.messages.created')
+    $scope.submit = FormService.submit $scope, $scope.comment,
+      flashSuccess: 'comment_form.messages.created'
+      successCallback: =>
+        $scope.comment.body = ''
```

Comment:
Hm, I did try to stick with that method and wasn't able to get it to go, the intuitive '$scope.comment = Records.comments.build' implementation has the awkward behaviour of clearing the comment form, but then re-posting your first comment when you submit again. 

I'm pretty sure it's because FormService retains a reference to the $scope.comment which was passed in, so changing the comment on the scope doesn't affect the comment the FormService is looking at.

This isn't a problem in other places because our forms are mostly in modals, so resetting the model afterwards isn't a thing.

Maybe we need to expose a method on the form service to reset the model it's looking at somehow?